Replace PageableManagerInterface in favor for PageableInterface (#1916)
sonata-project/SonataMediaBundle: UPGRADE-3.x.md

UPGRADE FROM 3.x to 3.x
=======================

### Remove deprecated calls

> - `Sonata\Doctrine\Model\PageableManagerInterface` is no longer used in profit of `Sonata\DatagridBundle\Pager\PageableInterface`

Reviewer: ```suggestion
- `Sonata\Doctrine\Model\PageableManagerInterface` is no longer used in profit of `Sonata\DatagridBundle\Pager\PageableInterface`.
```

> ### Remove deprecated calls
> 
> - `Sonata\Doctrine\Model\PageableManagerInterface` is no longer used in profit of `Sonata\DatagridBundle\Pager\PageableInterface`.

Reviewer: ```suggestion
### Sonata\DatagridBundle\Pager\PageableInterface

- Usages of `Sonata\Doctrine\Model\PageableManagerInterface` were replaced in favor of `Sonata\DatagridBundle\Pager\PageableInterface`.
```

### Dependencies

> - drop support for `sonata-project/datagrid-bundle` < ^3.0.

Reviewer: ```suggestion
- drop support for `sonata-project/datagrid-bundle` < 3.0.
```

> - drop support for `sonata-project/datagrid-bundle` < 3.0.

Reviewer: ```suggestion
- Drop support for `sonata-project/datagrid-bundle` < 3.0.
```

>     If you are extending these method you MUST add argument and return type declarations:

Reviewer: BTW, we always consider this kind of change as a BC break, thus this PR should target `master`.
```suggestion
    If you are extending these methods, you MUST add argument and return type declarations:
```

    - `Sonata\MediaBundle\Entity\GalleryManager::getPager()`
    - `Sonata\MediaBundle\Entity\MediaManager::getPager()`
    - `Sonata\MediaBundle\Model\MediaManager::getPager()`

>     - `Sonata\MediaBundle\Model\MediaManager::getPager()`

Reviewer: Why is this method mentioned twice?

Author: One if for entity, one is for document.

Reviewer 2: Should be ``Sonata\MediaBundle\Document\MediaManager::getPager()`` then ?

There is no `Sonata\MediaBundle\Model\MediaManager`

Author: Oh, correct. I check it in phone and miss mediamanager is twice

Author: Done.
